powertools-kafka/src/test/java/software/amazon/lambda/powertools/kafka/serializers/KafkaProtobufDeserializerTest.java

```diff
@@ -140,8 +162,7 @@ private byte[] createProtobufDataWithComplexMessageIndex(TestProduct product) th
 
         // Write complex message index array [1,0]
         codedOutput.writeUInt32NoTag(2); // Array length
-        codedOutput.writeUInt32NoTag(1); // First index value
-        codedOutput.writeUInt32NoTag(0); // Second index value
+        codedOutput.writeUInt32NoTag(2); // First index value
```

Comment:
The length is no longer 2. What is the new rule we are following here?

This would remove one bytes of the protobuf data because it removes 2 bytes according to length prefix but only one byte is added manually here.